tools/flang2/flang2exe/llutil.cpp

```diff
@@ -2194,6 +2194,86 @@ metadata_args_need_struct(void)
   return ll_feature_metadata_args_struct(&llvm_get_current_module()->ir);
 }
 
+/**
+ * This function returns true for the types supported
+ * in function make_param_op
+ */
+bool
+should_preserve_param(SPTR sptr)
+{
+  bool should_preserve = false;
+  DTYPE dtype = DTYPEG(sptr);
+
+  switch (DTY(dtype)) {
+    case TY_ARRAY:
+    case TY_STRUCT:
```

Comment:
Are these two handled as part of this PR. If yes, please add a comment where else can they be dropped (also in process_params) from this PR

Reply:
Yes it is handled in the PR. I shall add the comment.